vdev/src/git.rs

```diff
@@ -63,10 +63,71 @@ pub fn get_modified_files() -> Result<Vec<String>> {
     Ok(capture_output(&args)?.lines().map(str::to_owned).collect())
 }
 
+pub fn set_config_values(config_values: &[(&str, &str)]) -> Result<String> {
+    let mut args = vec!["config"];
+
+    for (key, value) in config_values {
+        args.push(key);
+        args.push(value);
+    }
+
+    capture_output(&args)
+}
+
+/// Checks if the current directory's repo is clean
+pub fn check_git_repository_clean() -> Result<bool> {
+    Ok(Command::new("git")
+        .args(["diff-index", "--quiet", "HEAD"])
+        .stdout(std::process::Stdio::null())
+        .status()
+        .map(|status| status.success())?)
+}
+
+/// Commits changes from the current repo
+pub fn commit(commit_message: &str) -> Result<Output> {
+    let output = run_command_check_status(&["-am", commit_message])?;
+    Ok(output)
+}
+
+/// Pushes changes from the current repo
+pub fn push() -> Result<Output> {
+    let output = Command::new("git")
+        .arg("push")
+        .output()
+        .map_err(|e| anyhow!("{}", e))?;
+
+    Ok(output)
+}
+
+pub fn clone(repo_url: &str) -> Result<Output> {
+    let output = Command::new("git")
+        .arg("clone")
+        .arg(repo_url)
+        .output()
+        .map_err(|e| anyhow!("{}", e))?;
+
+    Ok(output)
+}
+
 fn capture_output(args: &[&str]) -> Result<String> {
     Command::new("git").in_repo().args(args).capture_output()
 }
 
+// TODO: Potentially modify capture_output function with this implementation if it satisfies the use case
+fn run_command_check_status(args: &[&str]) -> Result<Output> {
+    let mut command = Command::new("git");
+    command.args(args);
+
+    let output = command.output().context(format!("Failed to run command: git {:?}", args))?;
+    let status = output.status;
+    if !status.success() {
+        return Err(anyhow!("Command failed with exit code: {status}"));
+    }
+
+    Ok(output)
+}
```

Comment:
For this pull request, drop this wrapper and just call `.output` or `.capture_output` in the caller the same as the rest are doing, and move this check into your work on #16807.